oop-special_class_method-medium-q12 (#700)
canvas-gamification/canvas-gamification-junit-tests: src/oop/special_class_method/medium/q12/QuestionText.md

# oop-special_class_method-medium-q12

### Shirt Class:

> Your `Shirt` class already has the following items:

Reviewer: Please add back the question context.

# oop-special_class_method-medium-q12

### Question Text:

> Make a Java class called **Shirt** with the following description:

Reviewer 2: Customers love to shop online instead of driving to the store. Department stores need a system to model their current and past inventory so it can be displayed online. Use the requirements below to write a Java class `Shirt` to keep track of clothing in a store.

### Shirt Class:

Your `Shirt` class already has the following items:

- Attributes:
    - An `int` called **size**
    - An `int` called **yearMade**
    - These are not visible from other classes

- Constructor:

>     - a `Shirt` constructor that is visible to other classes and accepts two `int` values and uses them to initialize

Reviewer 2: ```suggestion
    - a `Shirt` constructor that is visible to other classes and has two `int` parameters. The first initializes
```

>       the **size** and **yearMade** attributes.

Reviewer 2: ```suggestion
      the **size** attribute and the second the **yearMade** attribute.
```

You need to add the following items:

> - Methods:

Reviewer 2: make a note of these methods' visibility modifiers in their descriptions

>     - a `displaySize` method which return a `String` like `The size is x` where x is the value of **size** only if the

Reviewer 2: ```suggestion
    - a `displaySize` method which returns nothing and prints a `String` in the format of `The size is x` where x is the value of **size** only if the
```

      value of **yearMade** is greater than 2000. Otherwise it should print `This shirt is no longer carried`.

>     - a `getSize` method which returns the value stored in the **size** attribute

Reviewer 2: ```suggestion
    - a `getSize` method which returns the value stored in the **size** attribute.
```

# oop-special_class_method-medium-q12

## Question Text

Complete a Java class called **Shirt** with a constructor that initializes the attributes. Furthermore, the attribute called
**size** cannot be modified from outside the class, but it can be accessed through an accessor method. The class has

> another attribute called **yearMade** that cannot be modified or accessed from outside the class.

Reviewer 2: get rid of duplicate class information

### Shirt Class:

Your `Shirt` class already has the following items:

- Attributes:
    - An `int` called **size**
    - An `int` called **yearMade**
    - These are not visible from other classes

- Constructor:
    - a `Shirt` constructor that is visible to other classes and accepts two `int` and uses it to initialize
      the **size** and **yearMade** attributes.

You need to add the following items:

- Methods:
    - a `displaySize` method which return a `String` like `The size is x` where x is the value of **size** only if the

>       value of **yearMade** is greater than 2000.

Author: I just copied the question but I guess it is an initial modification for filtering a store database maybe?
